File: production/client/src/main/scala/kpn/client/common/map/Marker.scala
package kpn.client.common.map

import scala.scalajs.js

object Marker {

  def marker(nodeCoordinate: ol.Coordinate, color: String): ol.Feature = {
    val marker = new ol.Feature(ol.geom.Point(nodeCoordinate))
    marker.setStyle(Marker.style(color))
    marker
  }

  private def style(color: String): ol.style.Style = {
    ol.style.Style(
      image = ol.style.Icon(
        anchor = js.Array[Double](12, 41),
        src = "/assets/images/marker-icon-" + color + ".png"
      )
    )
  }
}

File: client/src/app/components/ol/domain/marker.ts
import {Coordinate} from "ol/coordinate";

import Feature from "ol/Feature";
import Point from "ol/geom/Point";
import Icon from "ol/style/Icon";
import IconAnchorUnits from "ol/style/IconAnchorUnits";
import Style from "ol/style/Style";

export class Marker {

  public static create(color: string, coordinate: Coordinate): Feature {
    const style = this.createStyle(color);
    const feature = new Feature(new Point(coordinate));
    feature.setStyle(style);
    return feature;
  }

  private static createStyle(color: string): Style {
    const src = `assets/images/marker-icon-${color}.png`;
    return new Style({
      image: new Icon({
        anchor: [12, 41],
        anchorXUnits: IconAnchorUnits.PIXELS,
        anchorYUnits: IconAnchorUnits.PIXELS,
        src: src
      })
    });
  }

}
